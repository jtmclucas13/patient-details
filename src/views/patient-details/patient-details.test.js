import React from 'react';
import { shallow } from 'enzyme';
import PatientDetails from './patient-details';

describe('component PatientDetails', () => {
    test('renders as expected without whisker plot', () => {
        const wrapper = renderPatientDetails();

        expect(wrapper).toMatchSnapshot();
    });

    test('renders with whisker plot', () => {
        const allWeights = [];
        const weight = 123;

        const wrapper = renderPatientDetails({ allWeights, weight });

        expect(wrapper).toMatchSnapshot();
    });
});

/**
* Helper function for rendering PatientDetails in tests without boilerplate required props
* @param {Object} [props={}] props for PatientDetails
* @return {ShallowWrapper} enzyme wrapper for PatientDetails 
*/
function renderPatientDetails(props = {}) {
    const propsToUse = {
        dob: '1/1/2018',
        name: {},
        ...props,
    };

    return shallow(<PatientDetails {...propsToUse} />);
}
