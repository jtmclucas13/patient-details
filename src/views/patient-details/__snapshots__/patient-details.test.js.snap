// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component PatientDetails renders as expected without whisker plot 1`] = `
<div
  className="patient-details"
>
  <h2>
    Patient Details - 
     
  </h2>
  <div
    className="patient-details__identifying-info"
  >
    <span>
      MRN: 
    </span>
    <span>
      1/1/2018
    </span>
    <span>
      0
       year old 
    </span>
  </div>
  <div
    className="patient-details__tumor-info"
  >
    cm 
    , 
  </div>
  <div
    className="patient-details__weight"
  >
     lbs
  </div>
  <Link
    replace={false}
    to="/"
  >
    <button
      className="patient-details__back"
    >
      Patient List
    </button>
  </Link>
</div>
`;

exports[`component PatientDetails renders with whisker plot 1`] = `
<div
  className="patient-details"
>
  <h2>
    Patient Details - 
     
  </h2>
  <div
    className="patient-details__identifying-info"
  >
    <span>
      MRN: 
    </span>
    <span>
      1/1/2018
    </span>
    <span>
      0
       year old 
    </span>
  </div>
  <div
    className="patient-details__tumor-info"
  >
    cm 
    , 
  </div>
  <div
    className="patient-details__weight"
  >
    123
     lbs
    <WhiskerPlot
      allValues={Array []}
      className="patient-details__weight-plot"
      targetValue={123}
      width={200}
    />
  </div>
  <Link
    replace={false}
    to="/"
  >
    <button
      className="patient-details__back"
    >
      Patient List
    </button>
  </Link>
</div>
`;
